--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,4 +1,5 @@
 module(
     name = "freetype",
     version = "2.13.3.bcr.1",
 )
+bazel_dep(name = "libpng", version = "1.6.44")
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -55,8 +55,10 @@
     copts = [
         "-DFT2_BUILD_LIBRARY",
         "-UDEBUG",
+        "-DFT_CONFIG_OPTION_USE_PNG",
     ],
     includes = ["include"],
     textual_hdrs = glob(["src/**/*.c"]),
+    deps = ["@libpng"],
     visibility = ["//visibility:public"],
 )
