diff --git a/imgui_widgets.cpp b/imgui_widgets.cpp
--- a/imgui_widgets.cpp
+++ b/imgui_widgets.cpp
@@ -2678,3 +2678,3 @@ bool ImGui::DragScalar(const char* label, ImGuiDataType data_type, void* p_data,
     if (g.LogEnabled)
         LogSetNextTextDecoration("{", "}");
-    RenderTextClipped(frame_bb.Min, frame_bb.Max, value_buf, value_buf_end, NULL, ImVec2(0.5f, 0.5f));
+    RenderTextClipped(frame_bb.Min, frame_bb.Max, value_buf, value_buf_end, NULL, ImVec2(0.0f, 0.5f));
