diff --git a/c/build.rs b/c/build.rs
index def89d3..aebdcde 100644
--- a/c/build.rs
+++ b/c/build.rs
@@ -13,8 +13,10 @@ use std::env;
 use std::fs;
+use std::path::Path;
 
 fn main() {
-    fs::create_dir_all("build/include/pathfinder").expect("Failed to create directories!");
+    let out_dir = env::var_os("OUT_DIR").unwrap();
+    fs::create_dir_all(Path::new(&out_dir).join("build/include/pathfinder")).expect("Failed to create directories!");
     let crate_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
     cbindgen::generate(crate_dir).expect("cbindgen failed!")
-                                 .write_to_file("build/include/pathfinder/pathfinder.h");
+                                 .write_to_file(Path::new(&out_dir).join("build/include/pathfinder/pathfinder.h"));
 }
